crates/wasmtime/src/config.rs

```diff
@@ -2203,6 +2220,36 @@ impl Config {
         self.detect_host_feature = Some(detect);
         self
     }
+
+    /// Configures Wasmtime to not use host-based trap handlers, for example
+    /// disables signal-handler registration on Unix platforms.
+    ///
+    /// Wasmtime will by default leverage host-based trap handlers to make
+    /// generated code more efficient. For example an out-of-bounds load in
+    /// WebAssembly will result in a segfault that is caught by a signal handler
+    /// in Wasmtime by default. Another example is divide-by-zero is reported by
+    /// hardware rather than explicitly checked and Wasmtime turns that into a
+    /// trap. Registering signal handlers can be a portability hazard though in
+    /// some environments so this option exists to disable this behavior.
+    ///
+    /// When host-based trap handlers are disabled then generated code will
+    /// never rely on segfaults for example. Generated code will be slower
+    /// because bounds checks must be explicit along with other operations like
+    /// integer division which must check for zero.
+    ///
+    /// When this option is disable it additionally requires that the
+    /// `enable_heap_access_spectre_mitigation` and
+    /// `enable_table_access_spectre_mitigation` Cranelift settings are
+    /// disabled. This means that generated code must have spectre mitigations
+    /// disabled. This is because spectre mitigations rely on faults from
+    /// loading from the null address to implement bounds checks.
+    ///
+    /// This option defaults to `true` meaning that host trap handlers are
+    /// enabled by default.
+    pub fn host_trap_handlers(&mut self, enable: bool) -> &mut Self {
+        self.tunables.host_trap_handlers = Some(enable);
+        self
+    }
```

Comment:
Yeah, expanding on my previous naming bikeshed, I think phrasing/renaming this stuff as `virtual_memory_traps` or maybe `signals_based_traps` will clarify things for users a lot here.

The latter is slightly more precise in that a `ud2` isn't actually related to virtual memory, but opens the can of worms that is "windows has something else that is similar to signals but not actually called signals" I think.

The other thing that might be good to flush out a tiny bit more here is the portability hazard bit and say something about which environments might want to disable these handlers. Even just adding something like "for example, embedded operating systems that do not support virtual memory" in parens after "can be a portability hazard in some environments" is probably good enough.